### PR TITLE
Solve unstable inconsistencies in SPI master driver

### PR DESCRIPTION
**Pull request by JurajSadel**
closes https://github.com/esp-rs/esp-hal/issues/2793

This is the result mostly of trial and error and fixing errors/warnings so I might have missed something here :D

### CODE REVIEW

**Comment by bugadani** [comment]
Just one more thing: if `half_duplex_read`/write are unstable, Address and Command can be, too, I think.

**Review by bjoernQ — APPROVED** [review]
LGTM

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks!

File: esp-hal/src/spi/master.rs [hunk 1]
```diff
@@ -1904,18 +1938,19 @@ mod dma {
...
-            cmp::min,
-            ops::{Deref, DerefMut},
-        };
+        #[cfg(any(doc, feature = "unstable"))]
+        use core::cmp::min;
+        use core::ops::{Deref, DerefMut};
 
         use super::*;
 
+        #[cfg_attr(not(feature = "unstable"), allow(dead_code))]
```
Comment (jessebraham) [inline esp-hal/src/spi/master.rs:1947]: If this is only used when `unstable` is enabled, shouldn't it just be cfg'd away entirely when the feature is disabled?

Comment (bugadani) [reply esp-hal/src/spi/master.rs:1947]: The problem is that `#[instability::unstable]` marks things `pub(crate)`, and doesn't remove them. If related things are removed completely, the project may end up not compiling.

**Pull request merged**